File: flutter_app/lib/app/core/network/models/start_session_request.dart
/// START_SESSION 请求模型。
class StartSessionRequest {
  final String sessionId;
  final int playerCount;
  final String gameMode;
  final bool randomizeFirstPlayer;

  const StartSessionRequest({
    required this.sessionId,
    required this.playerCount,
    required this.gameMode,
    required this.randomizeFirstPlayer,
  });

  Map<String, dynamic> toJson() {
    return <String, dynamic>{
      'sessionId': sessionId,
      'playerCount': playerCount,
      'gameMode': gameMode,
      'randomizeFirstPlayer': randomizeFirstPlayer,
    };
  }
}

File: src/main/java/com/monopoly/dto/StartSessionRequest.java
package com.monopoly.dto;

/**
 * START_SESSION payload: players, HVM/PVP/LLM/AI_VS_AI, AI difficulty, first player.
 */
public class StartSessionRequest {

    private String sessionId;
    /** 2–5 */
    private int playerCount;
    /** HVM, PVP, LLM, or AI_VS_AI. */
    private String gameMode;
    /** EASY / NORMAL / HARD; used by HVM sessions. */
    private String aiDifficulty;
    private boolean randomizeFirstPlayer;

    public StartSessionRequest() {
    }

    public String getSessionId() {
        return sessionId;
    }

    public void setSessionId(String sessionId) {
        this.sessionId = sessionId;
    }

    public int getPlayerCount() {
        return playerCount;
    }

    public void setPlayerCount(int playerCount) {
        this.playerCount = playerCount;
    }

    public String getGameMode() {
        return gameMode;
    }

    public void setGameMode(String gameMode) {
        this.gameMode = gameMode;
    }

    public String getAiDifficulty() {
        return aiDifficulty;
    }

    public void setAiDifficulty(String aiDifficulty) {
        this.aiDifficulty = aiDifficulty;
    }

    public boolean isRandomizeFirstPlayer() {
        return randomizeFirstPlayer;
    }

    public void setRandomizeFirstPlayer(boolean randomizeFirstPlayer) {
        this.randomizeFirstPlayer = randomizeFirstPlayer;
    }
}
